### ISSUE TITLE
How to create Directional graph with Graphframe

### ISSUE THREAD

**Issue by mavencode01** [graphframes/graphframes#176]
I'm trying to run the connected component algorithm to cluster my dataset using a directional graph (similar to a recursive query).

Is there any way to create a directional graph such that the connected component only transverse in one direction? So far, I haven't seen how to in your documentation. @felixcheung any idea please?

**Comment by felixcheung** [comment]
are you looking for https://en.wikipedia.org/wiki/Strongly_connected_component?

**Comment by mavencode01** [comment]
@felixcheung I don't think SCC can be used here (correct me if I'm wrong) since it requires every node to be reachable from other nodes. But basically, I need to do a recursion like SQL recursive query.

For example:
````
   a -> b 
   b -> c
   m -> b
````

Strongly CC:
```
Set = {a, b}
Set = {b, c}
Set = {m, b}
```

Connected component:
 It will be the set = ```{a, b, d, m}``` -- notice and `m` connected to the node`a -> b -> m` too

But in recursion
```
a -> b -> c = {a, b, c}
and
m -> b = {m, b}
this give the two sets  {a, b, c} and {m, b}
```

**Comment by mavencode01** [comment]
correct, it needs to avoid loops too. I'm wondering if Pregel can do this?

**Comment by SemyonSinchenko** [comment]
> I'm wondering if Pregel can do this?

At the moment we are investigating the possibility to provide in Pregel support for both directed and undirected cases (by adding sendMsgToSrc along with sendMsgToDst). I'm closing it now in favor of #556
